### DISCUSSION TITLE
下载问题（FFMPEG，线程下载器）

### DISCUSSION THREAD

**Discussion by blacktime1999** [EchterAlsFake/unofficial-api-for-pornhub#26] [Q&A] [answered]
有没有快速下载的方法

**Comment by blacktime1999** [comment]
> 有没有快速下载的方法

用了FFMPEG，效率并不高

**Comment by Egsagon** [comment, marked as answer]
Hi,
The FFMPEG downloader is close to be the fatest way to download a video.
The download speed depends on the quality of your connection and the size of the video.

If you want the download to be faster, you might want to:
- Try to make your internet connection better,
- Download videos in lower quality,
- If you use a VPN, select a server closer to your country,
- Inject your own FFMPEG settings by overidding `phub.consts.FFMPEG_COMMAND`,
- Use the `threaded` downloader

Alternatively, if you want to use a 3rd party downloader (for exemple a threaded download manager), you can refer to the [documentation](https://phub.readthedocs.io/en/latest/guides/download.html#even-more-advanced-downloading) for advanced downloading and retrieving the video M3U file, which is much faster to download than a whole video. You might want to dig towards M3U8 file downloaders. There are a bunch of them available as python packages, and probably as apps too.
